### PR TITLE
Fix getTransfersForX() bug

### PR DESCRIPTION
**Pull request by thebrianchen**
Fixed a bug with `getTransfersForOwner()` and `getTransfersForContract()` methods in the `NftNamespace`, where some NFTs would not be returned if the NFT was transferred multiple times.

`getNftMetdataBatch()` de-dupes tokens. This caused a bug where the if a batch contained the same token twice, the second token would not be hydrated with nft metadata.

Changes:
- Create a mapping of each token to the underlying NFT that is used to hydrate the transfer metadata
- Added an integration test to verify the behavior
- Deleted extra integration test that was a duplicate.

### CODE REVIEW

File: src/internal/nft-api.ts [hunk 1]
```diff
@@ -798,10 +798,27 @@ async function getNftsForTransfers(
...
+    const key = `${nft.contract.address}-${BigNumber.from(
+      nft.tokenId
+    ).toString()}`;
+    nftsByTokenId.set(key, nft);
+  });
+
+  const transferredNfts = metadataTransfers.map(t => {
+    const key = `${t.token.contractAddress}-${BigNumber.from(
+      t.token.tokenId
+    ).toString()}`;
```
Comment (brg8) [inline src/internal/nft-api.ts:816]: Is it always the case that the `nfts` list will contain every NFT in the `metadataTransfers` list?

**Review by brg8 — APPROVED** [review]
